src/Component.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProcessorIconv;
+
+use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class Component extends BaseComponent
+{
+    public function run(): void
+    {
+        /** @var Config $config */
+        $config = $this->getConfig();
+        try {
+            iconv($config->getSourceEncoding(), 'UTF-8', 'abc');
+        } /** @noinspection PhpRedundantCatchClauseInspection */ catch (\ErrorException $e) {
+            throw new UserException('Source encoding ' . $config->getSourceEncoding() . ' is invalid.');
+        }
+        try {
+            iconv('UTF-8', $config->getTargetEncoding(), 'abc');
+        } /** @noinspection PhpRedundantCatchClauseInspection */ catch (\ErrorException $e) {
+            throw new UserException('Target encoding ' . $config->getTargetEncoding() . ' is invalid.');
+        }
+        $filter = 'convert.iconv.' . $config->getSourceEncoding() .  '/' .
```

Comment:
To použití filtrů je na čtení docela komplikovaný, nestačil by cli obalenej `Symfony\Process`? Má to nějaký výhody kromě toho, že to dokáže líp identifikovat chyby?

Reply:
me prijde to, ze to dokaze lip detekovat chyby jako docela dobrej duvod, ne? :)

Reply:
Asi jo no :-)